pkg/controller/nebulascheduledbackup/nebula_scheduled_backup_manager.go

```diff
@@ -0,0 +1,250 @@
+
+	// Detect if schedule for nebula scheduled backup is changed
+	if scheduledBackup.Status.CurrSchedule != scheduledBackup.Spec.Schedule {
+		klog.Infof("Schedule change detected for scheduled backup %v/%v", scheduledBackup.Namespace, scheduledBackup.Name)
+		var custTimeToUse metav1.Time
+		if scheduledBackup.Status.LastBackupTime == nil {
+			custTimeToUse = *scheduledBackup.Status.LastBackupTime
+		} else {
+			custTimeToUse = metav1.Now()
+		}
+
+		nextBackupTime, err := calculateNextRunTime(scheduledBackup, &custTimeToUse)
+		if err != nil {
+			return fmt.Errorf("calculating next runtime for nebula scheduled backup %s/%s failed. err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+		}
+
+		if err := bm.clientSet.NebulaScheduledBackup().UpdateNebulaScheduledBackupStatus(scheduledBackup, &kube.ScheduledBackupUpdateStatus{
+			CurrSchedule:   scheduledBackup.Spec.Schedule,
+			NextBackupTime: nextBackupTime,
+		}); err != nil {
+			return fmt.Errorf("update nebula scheduled backup %s/%s status err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+		}
+		return utilerrors.ReconcileErrorf("Schedule for scheduled backup [%s/%s] updated successfully to %s.", scheduledBackup.Namespace, scheduledBackup.Name, scheduledBackup.Spec.Schedule)
+	} else if scheduledBackup.Status.Phase == v1alpha1.ScheduledBackupRunning {
+		klog.Infof("Running nebula backup detected for scheduled backup %v/%v", scheduledBackup.Namespace, scheduledBackup.Name)
+		// Detect if a running backup has completed or has failed
+		nebulaBackup, err := bm.clientSet.NebulaBackup().GetNebulaBackup(scheduledBackup.Namespace, fmt.Sprintf("%s-%s-%v", BackupPrefix, scheduledBackup.Name, scheduledBackup.Status.LastBackupTime.Unix()))
+		if err != nil {
+			return fmt.Errorf("getting running nebula backup for Nebula scheduled backup %s/%s failed. err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+		}
+
+		// Detect failed nebula backup
+		for _, condition := range nebulaBackup.Status.Conditions {
+			if (condition.Type == v1alpha1.BackupFailed || condition.Type == v1alpha1.BackupInvalid) && condition.Status == corev1.ConditionTrue {
+				if err := bm.clientSet.NebulaScheduledBackup().UpdateNebulaScheduledBackupStatus(scheduledBackup, &kube.ScheduledBackupUpdateStatus{
+					Phase: v1alpha1.ScheduledBackupJobFailed,
+				}); err != nil {
+					return fmt.Errorf("update nebula scheduled backup %s/%s status err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+				}
+				return utilerrors.ReconcileErrorf("nebula backup for scheduled backup [%s/%s] failed to complete. reason: %s message: %s", scheduledBackup.Namespace, scheduledBackup.Name, condition.Reason, condition.Message)
+			}
+		}
+
+		// Detect complete nebula backup
+		if condutil.IsBackupComplete(nebulaBackup) {
+			err := bm.clientSet.NebulaBackup().DeleteNebulaBackup(nebulaBackup.Namespace, nebulaBackup.Name)
```

Comment:
Why delete complete nebula backup?
I think it's should delete it when the backup is need no more (`MaxBackups` or `MaxReservedTime` ).

And the backup data should be deleted after the `Backup` cr deleted. 

What's your point? @MegaByte875 @kqzh

Reply:
Fixed